### PR TITLE
Let user specify first and last item to render + buffer

### PR DESCRIPTION
**Pull request by falky97**
Give the user more control but don't make any breaking changes.

### CODE REVIEW

**Comment by falky97** [comment]
Whoops I forgot to change `null` to `5` for the branches...

File: src/VirtualizedListRender.js [hunk 1]
```diff
@@ -25,15 +25,27 @@ export default {
         itemHeight: {
             type: Number,
             required: true
+        },
+        firstItemToRender: { // index of the first item to render
```
Comment (albertodeago) [inline src/VirtualizedListRender.js:29]: Why are you changing `firstItemToRender` and `lastItemToRender` from data to props?
Those variables are changed in the component when scrolling so they cannot be props.

Do you wanted to add a couple of props to select the "starting rendering state"? (I mean the list of items to render at start... I taken for granted that on the first render the firstItemToRender it's gonna be the first one)

**Comment by albertodeago** [comment]
Thanks for the MR btw

**Comment by albertodeago** [comment]
I'm also thinking of a different approach to solve the issue:
What if the component had a `update` method that you can call to "adjust" things when you change the height of the container or the list of items provided to the component?

Something like this
```vue
<template>
  <div id="app">
    <input type="text" v-model="listHeight">
    <div class="list-container" :style="{'height': listH}">
      <virtualized-list :items="list" :item-height="itemH" ref="vlist"> <!-- SAVE THE REF -->
        <template v-slot="data">
          <div class="item">
            static text - {{ data.content }}
          </div>
        </template>
      </virtualized-list>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    const fillArrayWithNumbers = n => Array.apply(null, Array(n)).map((x, i) => i);

    return {
      list: fillArrayWithNumbers(15000).map(i => ({id: i, content: "content-" + i})),
      itemH: 25,
      listHeight: 500 // initial height of the container
    }
  },
  computed: {
    listH() {
      return this.listHeight + "px";
    }
  },
  watch: {
    listH() {
      // recalculate the items to show in the virtual list
      this.$refs.vlist.update();
    }
  }
}
</script>
```

I'm also adding a `bench` prop to let the flexibility to choose the amount of element outside the view to render (I see it useful sometimes)

What do you think?

**Comment by falky97** [comment]
This looks like it could work. I ended up downloading this locally and making the bench a prop and it solves my problem.

But the update function also sounds really neat. I like that!

**Comment by albertodeago** [comment]
Ok, going to release an rc today (with both the bench and the update function), so you can use it directly from npm
Then I'll add the docs and soon release the version with the fix :)

**Comment by albertodeago** [comment]
You should be able to install the `0.1.0-rc.0` now and try out.
It prove the `update method` and the `bench` prop (number, defaults to 5 so no breaking changes).
Let me know if everything is working correctly

**Comment by falky97** [comment]
Just submitted PR #10 (your work) so I can start using this!! I have it working locally, but would prefer to use it directly from npm!!!

Thanks so much for your help on this. Once again, library is great. So fast and lightweight 😄

**Comment by albertodeago** [comment]
Done!
Thanks for the report and the using the component :)

**Pull request closed without merging**